frontend/installation-ui/web/micro-ui-internals/packages/modules/qc/src/components/FacilityDetails/Summary.js

```
@@ -6,32 +6,28 @@ import SingleRejectionReasonModal from "./SingleRejectionReasonModal";
   const [expanded, setExpanded] = useState(false);
   const [showRejectionModal, setShowRejectionModal] = useState(false);
   const rejectionReasons = useSelector((state) => state.qc.rejectionReasons);
-  const [rejectionData, setRejectionData] = useState(rejectionReasons?.[sectionName] || []);
   const [activeReasonId, setActiveReasonId] = useState(null);
   const dispatch = useDispatch();
   const selectedFacility = useSelector((state) => state.qc.common.selectedFacility);
+  const rejectionData = rejectionReasons?.[sectionName] || [];
 
   const handleSave = (data) => {
-    setRejectionData([...rejectionData, ...data?.filter((reason) => reason?.reason?.trim())]);
+    dispatch(setRejectionReasons(sectionName, [...rejectionData, ...data?.filter((reason) => reason?.reason?.trim())]));
```

Biome (2.1.2)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)